templates/travel/partials/util/initial-state.snip.html

```diff
@@ -0,0 +1,71 @@
+{{!
+<!---
+Copyright 2016 The AMP Start Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+}}
+
+{{!
+<!---
+  Property access (foo.bar) counts as an increase to amp-bind expression total
+  complexity. Instead of storing state in nested objects, all of the state
+  values are stored in the global scope and use underscores to denote "paths".
+  This helps avoid avoid the low complexity limit allowing for more elaborate
+  bind logic.
+
+  It unfortunately does require a ridiculous number of amp-state tags.
+-->
+}}
+
+<amp-state id="ui_hero"><script type="application/json">true</script></amp-state>
```

Comment:
@choumx - what do you think about this kind of usage?

Reply:
It's pretty bad. What expressions hit the complexity limit?

Reply:
@gifteconomist - can you answer Will's question here

Reply:
@choumx - This expression, which is used multiple times throughout the template, whenever we want to display dynamic response content: https://github.com/metalabdesign/ampstart/blob/cdc9a3f921bf567f6f73aeb70d87ab69714e236b/templates/travel/partials/util/amp-list-props.snip.html#L18-L26

Reply:
You can use a form with `amp-bind` for that. Get the response JSON from the form using an action and use it to render an amp-list with a `[src]` binding (it supports JSON as well as URL).

```html
<form on="submit-success:AMP.setState({myData: event.response})"></form>

<amp-list [src]="myData"></amp-list>
```

Reply:
I think @10xjs has experience experimenting with this

Reply:
We would definitely prefer not to return to the engineering phase for this aspect of the template. What we have is carefully crafted to achieve all of the desired functionality while staying within valid amp bounds. Any kind of fundamental change to the way the state management logic is set up could require major effort.

Reply:
Fair enough. 

@camelburrito Perhaps file a tracking issue for clean up/refactor later. I might relax the `amp-bind` complexity constraint a bit but using form seems like the right approach.